src/report.rs

```diff
@@ -0,0 +1,90 @@
+use std::{collections::HashSet, io::Write};
+
+use anyhow::Result;
+use thiserror::Error;
+use time::OffsetDateTime;
+
+/// A specific report of a file's checksum results.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct ChecksumReport {
+    file_path: String,
+    matching: bool,
+    got: String,
+}
+
+impl ChecksumReport {
+    pub fn new(file_path: String, matching: bool, got: String) -> Self {
+        Self {
+            file_path,
+            matching,
+            got,
+        }
+    }
+}
```

Comment:
Having a `new` constructor here feels a bit suspect. In general, you'd assume that a `ChecksumReport` is either created by:
- some owner, like a `Tool` or `App` struct, or
- a `Checksum`.

In this case, anyone can create a `ChecksumReport`, meaning mistakes could occur more often or the details within are too public.

I'd prefer to see something closer to a `Checksum` having a `checksum.get_report()`, as this only allows one API exposure for a `ChecksumReport` to be created. (this doesn't matter so much here, but it's a common consideration in larger Rust projects. C++ has similar standards here, so your experience there is super useful 😄)